cms/static/js/views/course_video_settings.js

```diff
@@ -0,0 +1,537 @@
+                if (!self.$el.is(event.target) && self.$el.has(event.target).length === 0) {
+                    self.closeCourseVideoSettings();
+                }
+            });
+        },
+
+        resetPlanData: function() {
+            this.selectedProvider = '';
+            this.selectedTurnaroundPlan = '';
+            this.selectedFidelityPlan = '';
+            this.availableLanguages = [];
+            this.activeLanguages = [];
+            this.selectedLanguages = [];
+        },
+
+        setActiveTranscriptPlanData: function() {
+            if (this.activeTranscriptionPlan) {
+                this.selectedProvider = this.activeTranscriptionPlan.provider;
+                this.selectedFidelityPlan = this.activeTranscriptionPlan.cielo24_fidelity;
+                this.selectedTurnaroundPlan = this.selectedProvider === 'Cielo24' ?
+                    this.activeTranscriptionPlan.cielo24_turnaround :
+                    this.activeTranscriptionPlan.three_play_turnaround;
+                this.activeLanguages = this.activeTranscriptionPlan.preferred_languages;
+            } else {
+                this.resetPlanData();
+            }
+        },
+
+        getTurnaroundPlan: function() {
+            var turnaroundPlan = null;
+            if (this.selectedProvider) {
+                turnaroundPlan = this.availableTranscriptionPlans[this.selectedProvider].turnaround;
+            }
+            return turnaroundPlan;
+        },
+
+        getFidelityPlan: function() {
+            var fidelityPlan = null;
+            if (this.selectedProvider === 'Cielo24') {
+                fidelityPlan = this.availableTranscriptionPlans[this.selectedProvider].fidelity;
+            }
+            return fidelityPlan;
+        },
+
+        getPlanLanguages: function() {
+            var selectedPlan = this.availableTranscriptionPlans[this.selectedProvider];
+            if (this.selectedProvider === 'Cielo24') {
+                return selectedPlan.fidelity[this.selectedFidelityPlan].languages;
+            }
+            return selectedPlan.languages;
+        },
+
+        fidelitySelected: function(event) {
+            var $fidelityContainer = this.$el.find('.transcript-fidelity-wrapper');
+            this.selectedFidelityPlan = event.target.value;
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($fidelityContainer);
+
+            // Clear active and selected languages.
+            this.selectedLanguages = this.activeLanguages = [];
+            this.populateLanguages();
+        },
+
+        turnaroundSelected: function(event) {
+            var $turnaroundContainer = this.$el.find('.transcript-turnaround-wrapper');
+
+            this.selectedTurnaroundPlan = event.target.value;
+            // Remove any error if present already.
+            this.clearPreferanceErrorState($turnaroundContainer);
```

Comment:
Typo: `Preferance` --> `Preference`. You're using wrong spelling at several places.